File: work/jobs/Ruby_test/workspace/consumer.py
import re
import StringIO
import requests
import ConfigParser
from ConfigParser import SafeConfigParser
from xml.etree import ElementTree
from xml.etree.ElementTree import QName

rdf_namespace = 'http://www.w3.org/1999/02/22-rdf-syntax-ns#'
oslc_namespace = 'http://open-services.net/ns/core#'
dcterms_namespace = 'http://purl.org/dc/terms/'
test_case_namespace = 'http://some.thing/testcase#'
test_case_type = test_case_namespace + 'TestCase'

config_file = "consumer.ini"
testCase_uri = ''

config = ConfigParser.ConfigParser()
config.read(config_file)
resource_uri = config.get('DEFAULT','resource_uri')
destination = './'
file_name_prefix = config.get('DEFAULT','file_name_prefix')

r = requests.get(resource_uri)
xml= StringIO.StringIO( r.content )
root = ElementTree.parse(xml).getroot()

for node in root:
    for subnode in node:
        if subnode.tag == str( QName(  oslc_namespace ,'queryBase') ):
           resource_type = node.find(str( QName(oslc_namespace ,'resourceType')))
           resource = resource_type.get(str( QName(rdf_namespace ,'resource')))
           if resource == test_case_type:
               testCase_uri = subnode.get(str( QName(rdf_namespace ,'resource')))

r = requests.get(testCase_uri)
xml= StringIO.StringIO( r.content )
root = ElementTree.parse(xml).getroot()

for node in root:
    for subnode in node:
        if subnode.tag == str( QName(rdf_namespace ,'type')):
            resource = subnode.get(str( QName(rdf_namespace ,'resource')))
            if resource == test_case_type:
               identifier = node.find(str( QName(dcterms_namespace ,'identifier'))).text
               reqCoverage = node.find(str( QName(test_case_namespace ,'reqCoverage'))).text
               shortName = node.find(str( QName(dcterms_namespace ,'title'))).text
               purpose = node.find(str( QName(test_case_namespace ,'purpose'))).text
               description = node.find(str( QName(dcterms_namespace ,'description'))).text
               testType = node.find(str( QName(dcterms_namespace ,'type'))).text
               level = node.find(str( QName(test_case_namespace ,'level'))).text
               actionEvent = node.find(str( QName(test_case_namespace ,'actionEvent'))).text
               passCriteria = node.find(str( QName(test_case_namespace ,'passCriteria'))).text
               environmentRequirement = node.find(str( QName(test_case_namespace ,'environmentRequirement'))).text
               comment = node.find(str( QName(test_case_namespace ,'comment'))).text
               
               abstractTestCase = node.find(str( QName(test_case_namespace ,'abstractTestCase'))).text
               abstractTestCase = re.sub(r"Trace: [0-9]+/[0-9]+\n", "", abstractTestCase)
               f = open(destination + file_name_prefix + identifier + '.txt', 'w')
               f.write("[SHORTNAME=" + shortName.strip() +"]\n")
               f.write("\n#############################################\n")
               f.write("# [TestCaseSpecification]\n")
               f.write("# " + shortName.replace("\n", "\n# "))
               f.write("[RequirementSpecification]\n")
               f.write("# " + reqCoverage.replace("\n", "\n# "))
               f.write("[Purpose]\n")
               f.write("# " + purpose.replace("\n", "\n# "))
               f.write("[Description]\n")
               f.write("# " + description.replace("\n", "\n# "))
               f.write("[Type]\n")
               f.write("# " + testType.replace("\n", "\n# "))
               f.write("[Level]\n")
               f.write("# " + level.replace("\n", "\n# "))
               f.write("[ActionEvent]\n")
               f.write("# " + actionEvent.replace("\n", "\n# ").strip())
               f.write(" [PassCriteria]\n")
               f.write("# " + passCriteria.replace("\n", "\n# "))
               f.write("[EnvironmentRequirement]\n")
               f.write("# " + environmentRequirement.replace("\n", "\n# "))
               f.write("[Comment]\n")
               f.write("# " + comment.replace("\n", "\n# "))
               f.write("\n#############################################")
               f.write(abstractTestCase)
               f.close()

File: work/jobs/Ruby_test/workspace/testpa/linx4py/consumer.rb
require 'open-uri'
require 'xmlsimple'

resource_uri = 'http://localhost:8080/OSLC4JRegistry/catalog'
destination = './'
file_name_prefix = 'abstracttestcase_'
testCase_uri = ''

class Consumer
    def get_testCase()
        resource_uri = 'http://localhost:8080/OSLC4JRegistry/catalog'
        destination = './'
        file_name_prefix = 'abstracttestcase_'
        testCase_uri = ''
        conf_file = 'consumer.ini'
        f = File.open(conf_file, "r")
        while (line = f.gets)
            if line.match(/resource_uri = (.*)/)
                testCase_uri = line.match(/resource_uri = (.*)/)[1]
            end
            
            if line.match(/destination = (.*)/)
                destination = line.match(/destination = (.*)/)[1]
            end
            
            if line.match(/file_name_prefix = (.*)/)
                file_name_prefix =line.match(/file_name_prefix = (.*)/)[1]
            end
        end


        web_contents = open(resource_uri) {|f| f.read }
        data = XmlSimple.xml_in(web_contents)

        data["Description"].each do |description|
            if description.has_key?("resourceType")
                description["resourceType"].each do |resource_type|
                    if resource_type["rdf:resource"] == "http://some.thing/testcase#TestCase"
                        if description.has_key?("queryBase")
                            testCase_uri = description["queryBase"][0]["rdf:resource"]
                        end
                    end
                end
            end
        end
        testCase_uri
        
        web_contents = open(testCase_uri) {|f| f.read }
        data = XmlSimple.xml_in(web_contents)

        allTC = ''

        data["Description"].each do |description|
            if description["type"][0]["rdf:resource"] == "http://some.thing/testcase#TestCase"
                aTC = description["abstractTestCase"][0].sub!(%r"Trace: [0-9]+/[0-9]+\n", '')
                
                FileUtils.mkdir_p(destination) unless File.directory?(destination)
                ofname = "#{destination}#{file_name_prefix}#{description["identifier"][0]}.txt"
                f = File.open(ofname, "w")
                f.puts "#{aTC}"
                f.close
                allTC << aTC
            end
        end 

        loop do
            next if allTC.sub!(/^[ \t\f\r]+/o, '')  # Suppress spaces
            next if allTC.sub!(/^[ \t\f\r]*\n[ \t\f\r]*/o, '')  # Next line        
            break
        end
        allTC
    end
end
